pkg/cloudprovider/vm.go

```diff
@@ -77,8 +80,19 @@ func (c *VMClient) Get(rgName string, nodeName string) (compute.VirtualMachine,
 		}
 	}()
 
+	// Report errors if the client is throttled.
+	if c.retryAfterReader.After(time.Now()) {
+		return compute.VirtualMachine{}, fmt.Errorf("VMGet client throttled, retry after: %v", c.retryAfterReader)
+	}
+
 	vm, err := c.client.Get(ctx, rgName, nodeName, "")
 	if err != nil {
+		resp := vm.Response.Response
+		retryAfterDuration := getRetryAfter(resp)
+		if isThrottled(resp) || retryAfterDuration != 0 {
```

Comment:
We can probably merge `isThrottled` with `getRetryAfter` since every 429 requests should have a `Retry-After` header. So if a request is not being throttled, `getRetryAfter` will just return 0.